mitodl/micromasters
# Updated app to proxy requests through Nginx

#### What are the relevant tickets?
#1971 

#### What's this PR do?
Puts Nginx in front of uWSGI for both local dev and in Heroku

#### Where should the reviewer start?
config/nginx.conf, config/nginx.conf.erb, docker-compose.yml

#### How should this be manually tested?
Run docker-compose and visit the app at http://localhost:8080 and make sure that all app functions are working as expected.

#### Any background context you want to provide?
Using Nginx in front of uWSGI will allow us to properly handle chunked-content encoding without having to resort to arbitrarily long socket timeouts to accomodate users with high latency and/or unreliable network connections.

#### Screenshots (if appropriate)

#### What GIF best describes this PR or how it makes you feel?
http://giphy.com/gifs/sinking-this-is-fine-everything-l0HlMG1EX2H38cZeE

Proxying apps through Nginx solves the chunked-content issues that we
have been seeing with the profile image uploads. This commit updates
the docker-compose and procfiles to use Nginx in front of uWSGI for
serving all routes.
